dist/init/linux-upstart/caddy.conf

```diff
@@ -1,18 +1,25 @@
-description "Caddy startup script"
+description "Caddy HTTP/2 web server"
 author "Mathias Beke"
 
 start on runlevel [2345]
 stop on runlevel [016]
 
+console log
 
 setuid www-data
 setgid www-data
 
 respawn
 respawn limit 10 5
 
-limit nofile 4096 4096
+# Letsencrypt-issued certificates will be written to this directory.
+env HOME=/etc/ssl/caddy
+
+limit nofile 1048576 1048576
 
 script
-	exec /usr/bin/caddy -agree=true -conf=/etc/caddy/Caddyfile
-end script
\ No newline at end of file
+        mkdir -p /etc/caddy
+        rootdir=`mktemp -d -t "caddy-run.XXXXXX"`
+        cd /etc/caddy
```

Comment:
I'd personally go for `cd /`.  One of the main reasons is, if someone decides to `chroot(2)` the whole thing into whatever their env setup defaults are, they won't have to make unecessary changes or directories just to have things work...